site/src/pages/docs/server-graphql.mdx

```diff
@@ -0,0 +1,105 @@
+# Running a GraphQL service
+
+<Tip>
+
+Visit [armeria-examples](https://github.com/line/armeria-examples) to find a fully working example.
+
+</Tip>
+
+Let's assume we have the following [GraphQL](https://graphql.org/) service definition:
+
+```graphql
+type Query {
+  user(id: ID): User
+}
+
+type User {
+  id: ID
+  name: String
+}
+```
+
+Armeria provides a way to write an HTTP service using the GraphQL like this.
+
+
+## `DataFetcher`
+And to fetch the data you need an implementation like this:
+
+```java
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+class UserDataFetcher implements DataFetcher<User> {
+
+    private final Map<String, User> data = Map.of("1", new User("1", "hero"),
+                                                  "2", new User("2", "human"),
+                                                  "3", new User("3", "droid"));
+
+    @Override
+    public User get(DataFetchingEnvironment environment) throws Exception {
+        final String id = environment.getArgument("id");
+        return data.get(id);
+    }
+}
+```
+<Tip>
+
+Please note that this is entirely based on [graphql-java](https://www.graphql-java.com/).
+
+</Tip>
```

Comment:
Could we add simple explanations for schema and DataFetcher?

---

[GraphQL Java]((https://www.graphql-java.com/) is the Java (server) implementation for GraphQL.
This documentation shows how to serve a GraphQL Java over HTTP.
The main steps for creating a GraphQL service are:
1. Defining a GraphQL schema.
2. Implement how the actual data for a query is fetched.
3. Bind the scheme and data fetchers with `GraphqlService`.

## Defining Schema

A schema which defines each field that can be queried or mutated and what types those fields are.
This schema defines one top level field (in the type Query): which returns the details of a User.
```graphql
type Query {
  user(id: ID): User
}

type User {
  id: ID
  name: String
}
```

## Loading Data
A `DataFetcher` fetches data for one field while the query is executed.
The `UserDataFetcher` retrieves a user with a specific ID from a static map. This is made just for testing.
```
class UserDataFetcher implements DataFetcher<User> {
    ...
}
```

## Building `GraphqlService`

...

Reply:
I've added it mixed up with what I've written. 👍👍

Reply:
![localhost_8000_docs_server-graphql](https://user-images.githubusercontent.com/22594101/140028967-da5ed8fb-80fd-46b9-853d-b133520fae4e.png)